### PR TITLE
[vtk] Update to 9.1.0, Add qtquick feature

### LINKED ISSUE #23105: [VTK] update to <9.1.0>
Library name: ***VTK***

New version number: ***9.1.0***

Other information that may be useful (release notes, etc...)
https://github.com/Kitware/VTK/releases/tag/v9.1.0

### PR DESCRIPTION
**Pull request by raahilsha-z**
**Describe the pull request**

- #### What does your PR fix?  
Updates VTK, and enables qtquick via install vtk[qtquick]

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
Should be the same as 9.0.3

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
Yes

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

**Comment by Neumann-A** [comment]
Updating VTK requires updating:
- vtk-m
- cgns
- paraview

File: ports/vtk/FindLZ4.patch [hunk 1]
```diff
@@ -42,8 +51,10 @@ index 8c94e3bcd..ade3f9451 100644
 -  endif ()
 -endif ()
 +find_package(LZ4 CONFIG REQUIRED)
-+set_target_properties(lz4::lz4 PROPERTIES IMPORTED_GLOBAL TRUE)
++set_target_properties(lz4::lz4 PROPERTIES IMPORTED TRUE GLOBAL TRUE)
```
Comment (Neumann-A) [inline ports/vtk/FindLZ4.patch:54]: https://cmake.org/cmake/help/latest/prop_tgt/IMPORTED_GLOBAL.html

File: ports/vtk/FindExpat.patch [hunk 2]
```diff
@@ -1,5 +1,14 @@
+From b0615e9a48660d2a3ebc07da854d38316ff38880 Mon Sep 17 00:00:00 2001
+From: "DESKTOP-6M1EIQI\\Raahil" <raahil.sha@zimaging.io>
+Date: Wed, 2 Feb 2022 13:25:05 -0500
+Subject: [PATCH] FindExpat
+
+---
+ CMake/FindEXPAT.cmake | 34 ++++++++++++++++++++++++++++------
+ 1 file changed, 28 insertions(+), 6 deletions(-)
+
```
Comment (Neumann-A) [inline ports/vtk/FindExpat.patch:9]: Whats up with those unnecessary header/footer changes?

File: ports/vtk/portfile.cmake [hunk 3]
```diff
@@ -165,6 +159,8 @@ vcpkg_cmake_configure(
         # Select modules / groups to install
         -DVTK_USE_EXTERNAL:BOOL=ON
         -DVTK_MODULE_USE_EXTERNAL_VTK_gl2ps:BOOL=OFF # Not yet in VCPKG
+        -DVTK_MODULE_USE_EXTERNAL_VTK_ioss:BOOL=OFF # Not yet in VCPKG
+        -DVTK_MODULE_USE_EXTERNAL_VTK_cgns:BOOL=OFF # IOSS relies on this, but incompatible with the one currently in VCPKG
```
Comment (Neumann-A) [inline ports/vtk/portfile.cmake:163]: update cgns....

File: ports/vtk/FindExpat.patch (hunk 2 above)
Comment (raahilsha-z) [reply ports/vtk/FindExpat.patch:9]: I'm having Sourcetree generate the patch files for me - I can go back and manually remove these.

File: ports/vtk/FindLZ4.patch (hunk 1 above)
Comment (raahilsha-z) [reply ports/vtk/FindLZ4.patch:54]: I'm working with PCL, and found that IMPORTED_GLOBAL wasn't working when trying to build an app that uses PCL and VTK together. Changing to this fixed the issue.

File: ports/vtk/portfile.cmake (hunk 3 above)
Comment (raahilsha-z) [reply ports/vtk/portfile.cmake:163]: I'll update those 3 and submit new commits here.

File: ports/vtk/FindLZ4.patch (hunk 1 above)
Comment (Neumann-A) [reply ports/vtk/FindLZ4.patch:54]: then you can remove the line completely..... because the other properties are not used by cmake

**Comment by JonLiu1993** [comment]
@raahilsha-z ,Thanks for your pr, looks CI failed on many triplets:
This is error log, could you please take a look:

https://dev.azure.com/vcpkg/public/_build/results?buildId=67291&view=artifacts&pathAsName=false&type=publishedArtifacts
````
D:\buildtrees\paraview\src\67f1e7d191-e4a8dea14d.clean\VTKExtensions\FiltersRendering\vtkPVGeometryFilter.cxx(1360): error C2039: 'StructuredWithBlankingExecute': is not a member of 'vtkDataSetSurfaceFilter'
D:\installed\x86-windows\include\vtk-9.1\vtkDataSetSurfaceFilter.h(117): note: see declaration of 'vtkDataSetSurfaceFilter'
````

**Comment by JackBoosY** [comment]
Draft this PR since there is no response for one month.

**Comment by JackBoosY** [comment]
Closing this PR since it seems that no progress is being made. Please reopen if work is still being done.

**Comment by panda-z** [comment]
Any updates on this PR?

**Pull request closed without merging**